### ISSUE TITLE
event.preventDefault() not respected inside <Tabs.Tab>

### ISSUE THREAD

**Issue by peterbe** [mantinedev/mantine#2186]
### What package has an issue

@mantine/core

### Describe the bug

See demo link.

```jsx
        <Tabs.Tab value="messages" icon={<IconMessageCircle size={14} />}>
          Messages{' '}
          <a href="foo" onClick={(event) => event.preventDefault()}>
            X
          </a>
        </Tabs.Tab>
```
When I click on the text "Messages" I want that to be registered as a tab click. Just like browser tabs. 
But when I click the "X" I want to NOT switch to the "Messages" tab but instead write some NextJS router redirecting code that removes that tab. It kinda works, but clicking the "X" first causes the "Messages" tab to be selected, *then* it does the router stuff of changing which Tabs should be present. 

### What version of @mantine/hooks page do you have in package.json?

5.2.0

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/polished-sun-23fi3r?file=/src/App.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

I honestly don't know how to fix the bug, but with guidance willing to give it a try.

**Comment by rtivital** [comment]
It works as expected in your sandbox – default behavior is prevented, when anchor is clicked the url does not change.

**Comment by zhulien-ivanov** [comment]
You're looking for `event.stopPropagation`, not `event.preventDefault`.

**Comment by peterbe** [comment]
> It works as expected in your sandbox – default behavior is prevented, when anchor is clicked the url does not change.

The bug was that I initially had, say, "Gallery" as the active tab. But when I click on the "X" within the "Messages" tab, I did not want the "Messages" tab to be clicked/highlighted. 



> You're looking for `event.stopPropagation` instead of/in conjuction with `event.preventDefault`.

Good catch! Thank you. 
Fixed in https://codesandbox.io/s/vigorous-sound-lj2ctz?file=/src/App.tsx

Now, technically, I'll need to do more in that `onClick` on that `<a>` inside the `<Tabs.Tab>` but that's fine.
